src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,76 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { FormEvent, useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  addComment: (
+    newComment: CommentData,
+    offButtonLoad: (status: boolean) => void,
+    clearText: () => void,
+  ) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({ addComment }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [isNameEmpty, setIsNameEmpty] = useState(false);
+  const [isEmailEmpty, setIsEmailEmpty] = useState(false);
+  const [isTextEmpty, setIsTextEmpty] = useState(false);
+  const [isButtonLoading, setIsButtonLoading] = useState(false);
+
+  const clearText = () => setText('');
+
+  const onSubmitNewComment = (e: FormEvent) => {
+    e.preventDefault();
+    setIsButtonLoading(true);
+
+    if (!name) {
+      setIsNameEmpty(true);
+      setIsButtonLoading(false);
+    }
+
+    if (!email) {
+      setIsEmailEmpty(true);
+      setIsButtonLoading(false);
+    }
+
+    if (!text) {
+      setIsTextEmpty(true);
+      setIsButtonLoading(false);
+    }
+
+    if (name && email && text) {
+      const sendData = {
+        name,
+        email,
+        body: text,
+      };
+
+      addComment(sendData, setIsButtonLoading, clearText);
```

Comment:
Consider changing the `addComment` function signature to accept an object with named properties instead of multiple arguments. This will make it easier to understand and maintain the code.